src/json.js

```diff
@@ -95,13 +95,18 @@ export function getValueForExpr(obj, expr) {
       value = undefined;
       break;
     }
-    if (!isObject(value) ||
-            value[part] === undefined ||
-            !hasOwnProperty(value, part)) {
-      value = undefined;
-      break;
+    if (isArray(value) && arrayHasKey(value, part)) {
+      value = value[part];
+      continue;
     }
-    value = value[part];
+    if (isObject(value) &&
```

Comment:
great feed back y'all! I've added a commit that I hope addresses all of your comments